pinot-spi/src/main/java/org/apache/pinot/spi/data/FieldSpec.java

```diff
@@ -393,6 +393,7 @@ public enum DataType {
     BOOLEAN(INT, false, true),
     TIMESTAMP(LONG, false, true),
     STRING(false, true),
+    UNKNOWN(false, true),
```

Comment:
There are like one thousand places where FieldSpec.DataType, which is specially problematic in switch. Here the datatype I proposed in https://github.com/apache/pinot/pull/10361 may be useful, but we decided to drop it.

Have you checked the impact of this new literal in these switch? Or do you plan to do that in another PR?

Reply:
Yeah. this PR just adds the interface needed and handles the broker logic. 
Adding an extra type makes sure the code will error out when there is an error. instead of using some existing type to hide the error.

Reply:
My point is that this PR does add `UNKNOWN` literal into `FieldSpec.DataType`, but doesn't changes any of the places where `FieldSpec.DataType` was used. This is specially important on switches. If they include a `default` case, the new literal will fall on that case, which will usually throw exceptions. In case of `switch`s without default case it can even be worse.

Reply:
Oh. I''ll add the handling later since null literal isn't supported today anyway